File: security/src/main/java/net/yan100/compose/security/annotations/EnableJwtIssuer.java
package net.yan100.compose.security.annotations;


import net.yan100.compose.security.autoconfig.FileKeysRepositoryAutoConfiguration;
import net.yan100.compose.security.autoconfig.JwtIssuerAutoConfiguration;
import org.springframework.context.annotation.Import;

import java.lang.annotation.*;

/**
 * 启动 jwt 验证客户端
 *
 * @author TrueNine
 * @since 2022-12-14
 */
@Import({
  JwtIssuerAutoConfiguration.class,
  FileKeysRepositoryAutoConfiguration.class
})
@Inherited
@Documented
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface EnableJwtIssuer {
}

File: security/src/main/kotlin/net/yan100/compose/security/annotations/EnableJwtIssuer.kt
package net.yan100.compose.security.annotations

import net.yan100.compose.security.autoconfig.JwtIssuerAutoConfiguration
import org.springframework.context.annotation.Import
import java.lang.annotation.Inherited


/**
 * 启动 jwt 验证客户端
 *
 * @author TrueNine
 * @since 2022-12-14
 */
@Import(JwtIssuerAutoConfiguration::class)
@Inherited
@MustBeDocumented
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class EnableJwtIssuer
